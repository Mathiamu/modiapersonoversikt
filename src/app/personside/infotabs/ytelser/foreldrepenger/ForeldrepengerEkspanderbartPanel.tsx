import * as React from 'react';
import {
    Foreldrepengerettighet,
    getForeldepengerIdDato,
    getUnikForeldrepengerKey
} from '../../../../../models/ytelse/foreldrepenger';
import { formaterDato } from '../../../../../utils/stringFormatting';
import EkspanderbartYtelserPanel from '../felles-styling/EkspanderbartYtelserPanel';
import Foreldrepenger from './ForeldrePenger';
import { useInfotabsDyplenker } from '../../dyplenker';
import { useAppState, useOnMount } from '../../../../../utils/customHooks';
import { useDispatch } from 'react-redux';
import { toggleVisYtesle } from '../../../../../redux/ytelser/ytelserReducer';

interface Props {
    foreldrepenger: Foreldrepengerettighet;
}

function ForeldrepengerEkspanderbartpanel({ foreldrepenger }: Props) {
    const open = useAppState(state => state.ytelser.aapnedeYtesler).includes(foreldrepenger);
    const dispatch = useDispatch();
    const setOpen = (vis: boolean) => dispatch(toggleVisYtesle(foreldrepenger, vis));

    const dyplenker = useInfotabsDyplenker();
    useOnMount(() => {
        const erValgtIUrl = dyplenker.ytelser.erValgt(getUnikForeldrepengerKey(foreldrepenger));
        erValgtIUrl && setOpen(true);
    });

    const tittelTillegsInfo = [
        `ID-dato: ${formaterDato(getForeldepengerIdDato(foreldrepenger))}`,
        foreldrepenger.foreldrepengetype
    ];

    return (
        <EkspanderbartYtelserPanel
            open={open}
            setOpen={setOpen}
            tittel="Foreldrepenger"
            tittelTillegsInfo={tittelTillegsInfo}
        >
            <Foreldrepenger foreldrepenger={foreldrepenger} />
        </EkspanderbartYtelserPanel>
    );
}

export default ForeldrepengerEkspanderbartpanel;
