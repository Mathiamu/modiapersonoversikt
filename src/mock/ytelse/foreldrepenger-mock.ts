import * as faker from 'faker/locale/nb_NO';
import * as moment from 'moment';

import navfaker from 'nav-faker/dist/index';

import {
    Foreldrepengerperiode,
    ForeldrepengerResponse,
    Foreldrepengerrettighet
} from '../../models/ytelse/foreldrepenger';
import { getPeriode } from '../person/periodeMock';
import { fyllRandomListe } from '../utils/mock-utils';
import { getHistoriskUtbetaling, getKommendeUtbetaling } from './ytelse-utbetalinger-mock';
import { HistoriskUtbetaling, KommendeUtbetaling } from '../../models/ytelse/ytelse-utbetalinger';

export function getMockForeldrepenger(fødselsnummer: string): ForeldrepengerResponse {
    faker.seed(Number(fødselsnummer));
    navfaker.seed(fødselsnummer + 'foreldrepenger');

    if (navfaker.random.vektetSjanse(0.3)) {
        return {};
    }

    return {
        foreldrepenger: getForeldrepengerrettighet(fødselsnummer)
    };
}

function getForeldrepengerrettighet(fødselsnummer: string): Foreldrepengerrettighet {
    return {
        forelder: fødselsnummer,
        andreForeldersFnr: navfaker.personIdentifikator.fødselsnummer(),
        antallBarn: navfaker.random.integer(5),
        barnetsFødselsdato: moment(faker.date.recent()).format(moment.ISO_8601.__momentBuiltinFormatBrand),
        dekningsgrad: navfaker.random.integer(95),
        fedrekvoteTom: moment(faker.date.recent()).format(moment.ISO_8601.__momentBuiltinFormatBrand),
        mødrekvoteTom: moment(faker.date.recent()).format(moment.ISO_8601.__momentBuiltinFormatBrand),
        foreldrepengetype: 'FTYPE',
        graderingsdager: navfaker.random.integer(100),
        restDager: navfaker.random.integer(50),
        rettighetFom: moment(faker.date.recent()).format(moment.ISO_8601.__momentBuiltinFormatBrand),
        eldsteIdDato: moment(faker.date.recent()).format(moment.ISO_8601.__momentBuiltinFormatBrand),
        periode: fyllRandomListe<Foreldrepengerperiode>(() => getForeldrepengerperiode(fødselsnummer), 10)
    };
}

function getForeldrepengerperiode(fødselsnummer: string): Foreldrepengerperiode {
    return {
        fødselsnummer: fødselsnummer,
        harAleneomsorgFar: faker.random.boolean(),
        harAleneomsorgMor: faker.random.boolean(),
        arbeidsprosentMor: navfaker.random.integer(100),
        avslagsårsak: 'AVSLAG',
        avslått: 'AVSLÅTT',
        disponibelGradering: navfaker.random.integer(75),
        erFedrekvote: faker.random.boolean(),
        erMødrekvote: faker.random.boolean(),
        forskyvelsesårsak1: 'FÅRSAK1',
        forskyvelsesperiode1: getPeriode(),
        forskyvelsesårsak2: 'FÅRSAK2',
        forskyvelsesperiode2: getPeriode(),
        foreldrepengerFom: moment(faker.date.recent()).format(moment.ISO_8601.__momentBuiltinFormatBrand),
        midlertidigStansDato: moment(faker.date.recent()).format(moment.ISO_8601.__momentBuiltinFormatBrand),
        morSituasjon: faker.lorem.words(5),
        rettTilFedrekvote: 'RETTF',
        rettTilMødrekvote: 'RETTM',
        stansårsak: 'STANS',
        historiskeUtbetalinger: fyllRandomListe<HistoriskUtbetaling>(() => getHistoriskUtbetaling(faker), 5),
        kommendeUtbetalinger: fyllRandomListe<KommendeUtbetaling>(() => getKommendeUtbetaling(faker), 5)
    };
}