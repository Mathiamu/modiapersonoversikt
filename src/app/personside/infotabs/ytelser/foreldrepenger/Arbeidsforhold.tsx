import * as React from 'react';
import DescriptionList from '../felles-styling/DescriptionList';
import { FormatertKontonummer } from '../../../../../utils/FormatertKontonummer';
import { formaterDato } from '../../../../../utils/dateUtils';
import { formaterNOK } from '../../utbetalinger/utils/utbetalingerUtils';
import { AlertStripeInfo } from 'nav-frontend-alertstriper';
import { Arbeidsforhold } from '../../../../../models/ytelse/foreldrepenger';

function ArbeidsForhold({ arbeidsforhold }: { arbeidsforhold?: Arbeidsforhold }) {
    if (!arbeidsforhold) {
        return <AlertStripeInfo>Kunne ikke finne arbeidsforhold</AlertStripeInfo>;
    }

    const arbeidsSituasjonEntries = {
        Arbeidsgiver: arbeidsforhold.arbeidsgiverNavn,
        Kontonummer: arbeidsforhold.arbeidsgiverKontonr && (
            <FormatertKontonummer kontonummer={arbeidsforhold.arbeidsgiverKontonr} />
        ),
        Inntekstsperiode: arbeidsforhold.inntektsperiode,
        'Inntekt for perioden':
            arbeidsforhold.inntektForPerioden && 'NOK ' + formaterNOK(arbeidsforhold.inntektForPerioden),
        Refusjonstype: arbeidsforhold.refusjonstype,
        'Refusjon til dato': arbeidsforhold.refusjonTom && formaterDato(arbeidsforhold.refusjonTom),
    };

    return <DescriptionList entries={arbeidsSituasjonEntries} />;
}

export default ArbeidsForhold;
