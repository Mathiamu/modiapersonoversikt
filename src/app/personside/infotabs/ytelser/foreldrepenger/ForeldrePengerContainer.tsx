import * as React from 'react';
import { useEffect } from 'react';
import { loggEvent } from '../../../../../utils/frontendLogger';
import ForeldrepengerEkspanderbartpanel from './ForeldrepengerEkspanderbartPanel';
import RestResourceConsumer from '../../../../../rest/consumer/RestResourceConsumer';
import { ForeldrepengerResponse } from '../../../../../models/ytelse/foreldrepenger';

function ForeldrePengerContainer() {
    useEffect(() => {
        loggEvent('Sidevisning', 'Foreldrepenger');
    }, []);

    return (
        <RestResourceConsumer<ForeldrepengerResponse>
            spinnerSize="M"
            getRestResource={restResources => restResources.foreldrepenger}
        >
            {data => {
                if (!data.foreldrepenger || !data.foreldrepenger[0]) {
                    return null;
                }
                return data.foreldrepenger.map((foreldrepengerettighet, index) => (
                    <ForeldrepengerEkspanderbartpanel key={index} foreldrepenger={foreldrepengerettighet} />
                ));
            }}
        </RestResourceConsumer>
    );
}

export default ForeldrePengerContainer;
