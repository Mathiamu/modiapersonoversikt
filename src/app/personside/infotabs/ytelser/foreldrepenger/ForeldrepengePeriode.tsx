import * as React from 'react';
import { Foreldrepengerperiode } from '../../../../../models/ytelse/foreldrepenger';
import Utbetalinger from '../utbetalinger/Utbetalinger';
import styled from 'styled-components';
import theme from '../../../../../styles/personOversiktTheme';
import DescriptionList from '../felles-styling/DescriptionList';
import YtelserPeriode from '../felles-styling/YtelserPeriode';
import { formaterDato } from '../../../../../utils/dateUtils';

interface Props {
    periode: Foreldrepengerperiode;
    periodenr: number;
}

const Padding = styled.div`
    margin: ${theme.margin.px10} ${theme.margin.px20} ${theme.margin.px40};
`;

const Flex = styled.div`
    display: flex;
`;

const Big = styled.div`
    flex-basis: 55%;
`;

const Small = styled.div`
    flex-basis: 45%;
`;

export function convertBoolTilJaNei(verdi: boolean | null): string | null {
    switch (verdi) {
        case true:
            return 'Ja';
        case false:
            return 'Nei';
        default:
            return null;
    }
}

function ForeldrepengePeriode({ periode, periodenr }: Props) {
    const entries = {
        'Midlertidig stans': periode.midlertidigStansDato,
        Stansårsak: periode.stansårsak,
        Mødrekvote: convertBoolTilJaNei(periode.erMødrekvote),
        'Aleneomsorg Mor': convertBoolTilJaNei(periode.harAleneomsorgMor),
        'Rett til Mødrekvote': periode.rettTilMødrekvote,
        Fedrekvote: convertBoolTilJaNei(periode.erFedrekvote),
        'Aleneomsorg Far': convertBoolTilJaNei(periode.harAleneomsorgFar),
        'Rett til Fedrekvote': periode.rettTilFedrekvote,
    };
    return (
        <YtelserPeriode tittel={`Periode ${periodenr} - ${formaterDato(periode.foreldrepengerFom)}`}>
            <Flex>
                <Small>
                    <Padding>
                        <DescriptionList entries={entries} />
                    </Padding>
                </Small>
                <Big>
                    <Utbetalinger
                        kommendeUtbetalinger={periode.kommendeUtbetalinger}
                        historiskeUtbetalinger={periode.historiskeUtbetalinger}
                    />
                </Big>
            </Flex>
        </YtelserPeriode>
    );
}

export default ForeldrepengePeriode;
