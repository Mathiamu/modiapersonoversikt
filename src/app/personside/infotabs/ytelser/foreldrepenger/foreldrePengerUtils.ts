import { Foreldrepengerperiode, Foreldrepengerrettighet } from '../../../../../models/ytelse/foreldrepenger';
import { KommendeUtbetaling } from '../../../../../models/ytelse/ytelse-utbetalinger';
import { Periode } from '../../../../../models/periode';
import { dateComparator } from '../../../../../utils/dateUtils';
import { loggError } from '../../../../../utils/frontendLogger';
import { datoVerbose } from '../../utbetalinger/utils/utbetalingerUtils';

export function utledFraDatoForRettighet(foreldrepenger: Foreldrepengerrettighet): Date {
    return foreldrepenger.periode
        .map(periode => new Date(periode.foreldrepengerFom))
        .sort(dateComparator)[0];
}

function getAlleKommendeUtbetalinger(foreldrePenger: Foreldrepengerrettighet) {
    return foreldrePenger.periode
        .reduce(
            (acc: KommendeUtbetaling[], periode: Foreldrepengerperiode) =>
                [...acc, ...periode.kommendeUtbetalinger]
            ,
            []);
}

function getAlleVedtaksPerioder(alleKommendeUtbetalinger: KommendeUtbetaling[]) {
    return alleKommendeUtbetalinger.map(utbetaling => {
        if (!utbetaling.vedtak) {
            throw new Error('Kommende utbetaling i foreldrepengerettighet mangler vedtak');
        }
        return utbetaling.vedtak;
    });
}

export function utledMaksDato(foreldrePenger: Foreldrepengerrettighet): string {
    const alleKommendeUtbetalinger: KommendeUtbetaling[] = getAlleKommendeUtbetalinger(foreldrePenger);
    try {
        const alleVedtaksPerioder: Array<Periode> = getAlleVedtaksPerioder(alleKommendeUtbetalinger);
        const alleTilDatoerDecending =  alleVedtaksPerioder
            .map(periode => new Date(periode.til))
            .sort(dateComparator)
            .reverse();
        return datoVerbose(alleTilDatoerDecending[0]).sammensatt;
    } catch (error) {
        loggError(error, 'Mangler data for Ã¥ utlede maks dato', {alleKommendeUtbetalinger: alleKommendeUtbetalinger});
        return 'Mangler data';
    }
}