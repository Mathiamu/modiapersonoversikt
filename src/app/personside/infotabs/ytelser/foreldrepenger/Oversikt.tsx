import * as React from 'react';
import { Foreldrepengerrettighet } from '../../../../../models/ytelse/foreldrepenger';
import theme from '../../../../../styles/personOversiktTheme';
import { utledMaksDato, utledFraDatoForRettighet } from './foreldrePengerUtils';
import { datoVerbose } from '../../utbetalinger/utils/utbetalingerUtils';
import styled from 'styled-components';
import DescriptionList from '../DescriptionList';
import YtelserBullet from '../YtelserBullet';

interface Props {
    foreldrePenger: Foreldrepengerrettighet;
}

const Wrapper = styled.div`
    margin: 0 ${theme.margin.px20} ${theme.margin.px20};
  > *:not(:first-child) {
    border-top: ${theme.border.skilleDashed};
  }
`;

function Oversikt({foreldrePenger}: Props) {

    const foreldrePengeRetten = {
        Foreldrepengetype: foreldrePenger.foreldrepengetype,
        Dekningsgrad: foreldrePenger.dekningsgrad + '%',
        Dagsats: 'Ikke implementert',
        Graderingsdager: 'Ikke implementert',
        'Fedrekvote tilogmed': 'Ikke implementert',
        'Mødrekvote tilogmed': 'Ikke implementert',
        'Rettighet fra dato': datoVerbose(utledFraDatoForRettighet(foreldrePenger)).sammensatt,
        Restdager: foreldrePenger.restDager,
        Maksdato: utledMaksDato(foreldrePenger),
        Arbeidskategori: 'Ikke implementert'
    };

    const barnet = {
        Termindato: foreldrePenger.rettighetFom && datoVerbose(foreldrePenger.rettighetFom).sammensatt,
        Fødselsdato: foreldrePenger.barnetsFødselsdato && datoVerbose(foreldrePenger.barnetsFødselsdato).sammensatt,
        'Annen forelder': foreldrePenger.andreForeldersFnr,
        Omsorgsovertakelse: 'Ikke implementert',
        'Foreldre av samme kjønn': 'Ikke implementert',
        'Antall barn': foreldrePenger.antallBarn
    };

    return (
        <Wrapper>
            <YtelserBullet tittel="Om foreldrepengeretten">
                <DescriptionList entries={foreldrePengeRetten}/>
            </YtelserBullet>
            <YtelserBullet tittel="Om barnet">
                <DescriptionList entries={barnet}/>
            </YtelserBullet>
        </Wrapper>
    );
}

export default Oversikt;
