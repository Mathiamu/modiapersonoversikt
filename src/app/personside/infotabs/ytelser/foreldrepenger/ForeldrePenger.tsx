import * as React from 'react';
import { ForeldrepengerResponse } from '../../../../../models/ytelse/foreldrepenger';
import { AlertStripeInfo } from 'nav-frontend-alertstriper';
import ForeldrepengePeriode from './ForeldrepengePeriode';
import Oversikt from './Oversikt';
import { Undertittel } from 'nav-frontend-typografi';
import theme from '../../../../../styles/personOversiktTheme';
import styled from 'styled-components';

interface Props {
    foreldrePengerResponse: ForeldrepengerResponse;
}

const TittelStyle = styled.div`
  padding: ${theme.margin.px20} ${theme.margin.px20} ${theme.margin.px10};
`;

function Foreldrepenger(props: Props) {

    const foreldrepenger = props.foreldrePengerResponse.foreldrepenger;

    if (foreldrepenger === null) {
        return <AlertStripeInfo>Finner ingen foreldrepengerettigheter for bruker</AlertStripeInfo>;
    }

    const perioder = foreldrepenger.periode.map((periode, index) =>
        <ForeldrepengePeriode key={index} periode={periode}/>
    );

    return (
        <>
            <TittelStyle><Undertittel>Foreldrepenger</Undertittel></TittelStyle>
            <Oversikt foreldrePenger={foreldrepenger}/>
            <ul>
                {perioder}
            </ul>
        </>
    );
}

export default Foreldrepenger;
