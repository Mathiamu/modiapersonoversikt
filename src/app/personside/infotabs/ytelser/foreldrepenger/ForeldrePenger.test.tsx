import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { getForeldrepengerettighetMock } from '../../../../../mock/ytelse/foreldrepenger-mock';
import Foreldrepenger from './ForeldrePenger';
import { Foreldrepengerettighet } from '../../../../../models/ytelse/foreldrepenger';

test('Foreldrepengeperiode matcher snapshot', () => {
    const testRettighet = getForeldrepengerettighetMock('10108000398', 1234);
    const redusertRettighet: Foreldrepengerettighet = {
        ...testRettighet,
        arbeidsforhold: [testRettighet.arbeidsforhold[0], testRettighet.arbeidsforhold[0]],
        periode: [
            {
                ...testRettighet.periode[0],
                historiskeUtbetalinger: [testRettighet.periode[0].historiskeUtbetalinger[0]],
                kommendeUtbetalinger: [testRettighet.periode[0].kommendeUtbetalinger[0]],
            },
        ],
    };
    const result = renderer.create(<Foreldrepenger foreldrepenger={redusertRettighet} />);

    expect(result).toMatchSnapshot();
});
