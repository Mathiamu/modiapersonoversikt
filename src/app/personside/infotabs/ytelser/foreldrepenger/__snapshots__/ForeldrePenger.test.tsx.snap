// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Foreldrepengeperiode matcher snapshot 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c3 dt {
  color: #78706a;
}

.c3 dd {
  font-weight: bold;
  margin-top: 0.3rem;
}

.c3 > div {
  margin: 0.9375rem 0;
  padding-right: 1rem;
  width: 13rem;
  overflow-wrap: break-word;
}

.c12 {
  text-align: center;
}

.c8 {
  text-transform: uppercase;
}

.c7 {
  font-weight: bold;
}

.c10 {
  border-radius: 0.5rem;
  background-color: #e9e7e7;
  box-shadow: inset 0 0 0 0.3rem white,inset 0 0 0 0.35rem rgba(0,0,0,0.1);
  padding: 1.25rem;
}

.c6 > *:first-child {
  background-color: #cce1f3;
  padding: 0.2rem 1.25rem;
}

.c6 ol {
  padding: 0;
  margin: 0;
}

.c13 {
  overflow: auto;
  box-shadow: 0 0 0 0.05rem rgba(0,0,0,0.25);
  border-radius: .25rem;
  background-color: white;
}

.c13 table {
  text-align: left;
  width: 100%;
}

.c13 table thead {
  border-bottom: 0.2rem solid rgba(0,0,0,0.15);
  font-weight: bold;
}

.c13 table th,
.c13 table td {
  padding: 0.7rem;
  vertical-align: middle;
}

.c13 table th {
  padding-bottom: 0.7rem;
}

.c13 table tr:nth-child(even) {
  background-color: rgba(0,0,0,0.1);
}

.c11 {
  white-space: nowrap;
}

.c11 > *:not(:last-child) {
  margin-bottom: 1rem;
}

.c14 {
  padding: 0.5rem;
}

.c9 {
  padding: .8rem;
}

.c9 > *:not(:last-child) {
  margin-bottom: 2rem;
}

.c2 {
  border-radius: 0.5rem;
  background-color: #e9e7e7;
  box-shadow: inset 0 0 0 0.3rem white,inset 0 0 0 0.35rem rgba(0,0,0,0.1);
  padding: 1.25rem;
}

.c5 span:not(:last-child):after {
  content: '.';
}

.c4 li:not(:first-child) {
  border-top: solid 0.0625rem #c6c2bf;
}

.c0 {
  padding: .8rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 > * {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<article>
  <h2
    className="visually-hidden"
    tabIndex={-1}
  >
    Foreldrepengerettighet
  </h2>
  <div
    className="c0"
  >
    <div
      className="c1"
    >
      <div
        className="c2"
      >
        <h3
          className="typo-ingress"
        >
          Om foreldrepengeretten
        </h3>
        <dl
          className="c3"
        >
          <div>
            <dt
              className="typo-etikett-liten"
            >
              Foreldrepengetype
            </dt>
            <dd
              className="typo-normal"
            >
              Fødselspenger
            </dd>
          </div>
          <div>
            <dt
              className="typo-etikett-liten"
            >
              Dekningsgrad
            </dt>
            <dd
              className="typo-normal"
            >
              34%
            </dd>
          </div>
          <div>
            <dt
              className="typo-etikett-liten"
            >
              Rettighet fra dato
            </dt>
            <dd
              className="typo-normal"
            >
              29.04.2017
            </dd>
          </div>
          <div>
            <dt
              className="typo-etikett-liten"
            >
              Graderingsdager
            </dt>
            <dd
              className="typo-normal"
            >
              55
            </dd>
          </div>
          <div>
            <dt
              className="typo-etikett-liten"
            >
              Restdager
            </dt>
            <dd
              className="typo-normal"
            >
              41
            </dd>
          </div>
          <div>
            <dt
              className="typo-etikett-liten"
            >
              Maksdato
            </dt>
            <dd
              className="typo-normal"
            >
              —
            </dd>
          </div>
          <div>
            <dt
              className="typo-etikett-liten"
            >
              Arbeidskategori
            </dt>
            <dd
              className="typo-normal"
            >
              Arbeidstaker
            </dd>
          </div>
          <div>
            <dt
              className="typo-etikett-liten"
            >
              Fedrekvote til og med
            </dt>
            <dd
              className="typo-normal"
            >
              10.02.2019
            </dd>
          </div>
        </dl>
      </div>
      <div
        className="c2"
      >
        <h3
          className="typo-ingress"
        >
          Om barnet
        </h3>
        <dl
          className="c3"
        >
          <div>
            <dt
              className="typo-etikett-liten"
            >
              Termindato
            </dt>
            <dd
              className="typo-normal"
            >
              11.02.2019
            </dd>
          </div>
          <div>
            <dt
              className="typo-etikett-liten"
            >
              Fødselsdato
            </dt>
            <dd
              className="typo-normal"
            >
              11.02.2019
            </dd>
          </div>
          <div>
            <dt
              className="typo-etikett-liten"
            >
              Annen forelder
            </dt>
            <dd
              className="typo-normal"
            >
              03063850003
            </dd>
          </div>
          <div>
            <dt
              className="typo-etikett-liten"
            >
              Antall barn
            </dt>
            <dd
              className="typo-normal"
            >
              5
            </dd>
          </div>
        </dl>
      </div>
    </div>
    <div
      className="c2"
    >
      <h3
        className="typo-ingress"
      >
        Arbeidssituasjon
      </h3>
      <ol
        aria-label="Arbeidsgivere"
        className="c4"
      >
        <li>
          <dl
            className="c3"
          >
            <div>
              <dt
                className="typo-etikett-liten"
              >
                Arbeidsgiver
              </dt>
              <dd
                className="typo-normal"
              >
                Bakke - Solheim
              </dd>
            </div>
            <div>
              <dt
                className="typo-etikett-liten"
              >
                Kontonummer
              </dt>
              <dd
                className="typo-normal"
              >
                <span
                  className="c5"
                >
                  <span>
                    3517
                  </span>
                  <span>
                    91
                  </span>
                  <span>
                    78779
                  </span>
                </span>
              </dd>
            </div>
            <div>
              <dt
                className="typo-etikett-liten"
              >
                Inntekstsperiode
              </dt>
              <dd
                className="typo-normal"
              >
                Månedlig
              </dd>
            </div>
            <div>
              <dt
                className="typo-etikett-liten"
              >
                Inntekt for perioden
              </dt>
              <dd
                className="typo-normal"
              >
                40,062.00 NOK
              </dd>
            </div>
            <div>
              <dt
                className="typo-etikett-liten"
              >
                Refusjonstype
              </dt>
              <dd
                className="typo-normal"
              >
                Ikke refusjon
              </dd>
            </div>
            <div>
              <dt
                className="typo-etikett-liten"
              >
                Refusjon til dato
              </dt>
              <dd
                className="typo-normal"
              >
                11.02.2019
              </dd>
            </div>
            <div>
              <dt
                className="typo-etikett-liten"
              >
                Sykepenger fra og med
              </dt>
              <dd
                className="typo-normal"
              >
                10.02.2019
              </dd>
            </div>
          </dl>
        </li>
      </ol>
      <button
        className="knapp knapp--hoved"
        disabled={false}
        onClick={[Function]}
        type="submit"
      >
        Vis alle arbeidsforhold
      </button>
    </div>
  </div>
  <ol>
    <li
      className="c6"
    >
      <h3
        className="typo-normal"
      >
        <span
          className="c7"
        >
          <span
            className="c8"
          >
            Periode 1 - 29.04.2017
          </span>
        </span>
      </h3>
      <article
        className="c9"
      >
        <div
          className="c10"
        >
          <dl
            className="c3"
          >
            <div>
              <dt
                className="typo-etikett-liten"
              >
                Arbeidsprosent mor
              </dt>
              <dd
                className="typo-normal"
              >
                58%
              </dd>
            </div>
            <div>
              <dt
                className="typo-etikett-liten"
              >
                Mors situasjon
              </dt>
              <dd
                className="typo-normal"
              >
                nobis sit quo et quis
              </dd>
            </div>
            <div>
              <dt
                className="typo-etikett-liten"
              >
                Disponibel gradering
              </dt>
              <dd
                className="typo-normal"
              >
                9%
              </dd>
            </div>
            <div>
              <dt
                className="typo-etikett-liten"
              >
                Forskyvelsesperiode
              </dt>
              <dd
                className="typo-normal"
              >
                08.10.2014 - 22.08.2014
              </dd>
            </div>
            <div>
              <dt
                className="typo-etikett-liten"
              >
                Andre forskyvelsesperiode
              </dt>
              <dd
                className="typo-normal"
              >
                09.12.2014 - 26.08.2014
              </dd>
            </div>
            <div>
              <dt
                className="typo-etikett-liten"
              >
                Midlertidig stans
              </dt>
              <dd
                className="typo-normal"
              >
                2019-02-10
              </dd>
            </div>
            <div>
              <dt
                className="typo-etikett-liten"
              >
                Stansårsak
              </dt>
              <dd
                className="typo-normal"
              >
                Avsluttet
              </dd>
            </div>
            <div>
              <dt
                className="typo-etikett-liten"
              >
                Avslått
              </dt>
              <dd
                className="typo-normal"
              >
                Avslått
              </dd>
            </div>
            <div>
              <dt
                className="typo-etikett-liten"
              >
                Mødrekvote
              </dt>
              <dd
                className="typo-normal"
              >
                —
              </dd>
            </div>
            <div>
              <dt
                className="typo-etikett-liten"
              >
                Rett til Mødrekvote
              </dt>
              <dd
                className="typo-normal"
              >
                Ingen rett til mødrekvote
              </dd>
            </div>
            <div>
              <dt
                className="typo-etikett-liten"
              >
                Aleneomsorg Mor
              </dt>
              <dd
                className="typo-normal"
              >
                —
              </dd>
            </div>
            <div>
              <dt
                className="typo-etikett-liten"
              >
                Fedrekvote
              </dt>
              <dd
                className="typo-normal"
              >
                —
              </dd>
            </div>
            <div>
              <dt
                className="typo-etikett-liten"
              >
                Rett til Fedrekvote
              </dt>
              <dd
                className="typo-normal"
              >
                Rett til fedrekvote
              </dd>
            </div>
            <div>
              <dt
                className="typo-etikett-liten"
              >
                Aleneomsorg Far
              </dt>
              <dd
                className="typo-normal"
              >
                —
              </dd>
            </div>
          </dl>
        </div>
        <section
          className="c11"
        >
          <div
            className="c12"
          >
            <h4
              className="typo-undertittel"
            >
              Kommende utbetalinger
            </h4>
          </div>
          <div
            className="typo-normal c13"
          >
            <table
              role="table"
            >
              <thead
                role="rowgroup"
              >
                <tr
                  role="row"
                >
                  <th
                    role="columnheader"
                  >
                    Registeringsdato
                  </th>
                  <th
                    role="columnheader"
                  >
                    Type
                  </th>
                  <th
                    role="columnheader"
                  >
                    Periode
                  </th>
                  <th
                    role="columnheader"
                  >
                    Utbetalingsgrad
                  </th>
                  <th
                    role="columnheader"
                  >
                    Dagsats
                  </th>
                  <th
                    role="columnheader"
                  >
                    Bruttobeløp
                  </th>
                  <th
                    role="columnheader"
                  >
                    Arbeidsgiver
                  </th>
                  <th
                    role="columnheader"
                  >
                    Orgnummer
                  </th>
                  <th
                    role="columnheader"
                  >
                    Saksbehandler
                  </th>
                </tr>
              </thead>
              <tbody
                role="rowgroup"
              >
                <tr
                  role="row"
                >
                  <td
                    role="cell"
                  >
                    25.06.2018
                  </td>
                  <td
                    role="cell"
                  >
                    KONTOØVERFØRING
                  </td>
                  <td
                    role="cell"
                  >
                    14.01.2018 - 15.07.2017
                  </td>
                  <td
                    role="cell"
                  >
                    79%
                  </td>
                  <td
                    role="cell"
                  >
                    705.00 NOK
                  </td>
                  <td
                    role="cell"
                  >
                    466.00 NOK
                  </td>
                  <td
                    role="cell"
                  >
                    Vedvik - Lunde
                  </td>
                  <td
                    role="cell"
                  >
                    1234567890
                  </td>
                  <td
                    role="cell"
                  >
                    Mari Johannessen
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </section>
        <section>
          <div
            className="c14"
          >
            <div
              className="c12"
            >
              <h4
                className="typo-undertittel"
              >
                Utførte utbetalinger
              </h4>
            </div>
          </div>
        </section>
      </article>
    </li>
  </ol>
</article>
`;
